src/index.ts

```diff
@@ -121,7 +120,7 @@ export function parse(
     const key = str.slice(keyStartIdx, keyEndIdx);
 
     // only assign once
-    if (!__hasOwnProperty.call(obj, key)) {
+    if (obj[key] === undefined) {
```

Comment:
Can you double check performance? I didn’t change this because my numbers were worse doing the undefined check.

Reply:
Sure, let me bench it

Reply:
Bench LGTM, would you be free to rebase on main and remove the `if (value !== undefined)` now? Since it won't matter if `dec` returns `undefined` from https://github.com/jshttp/cookie/pull/179.